### PR TITLE
Two small fixes, one might have performance implications.

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
1) small vector::reserve wasn't updating capacity_ , meaning that large_store_ emplace was going to be N^2.  Also added a debug-only feature as reading the backing_store was driving me batty

2) return queue_index information as uint32_t which is consistent with the queue count in the queue family props

### CODE REVIEW

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: layers/vk_layer_data.h [hunk 1]
```diff
@@ -136,7 +136,7 @@ class small_vector {
     static const size_type kMaxCapacity = std::numeric_limits<size_type>::max();
     static_assert(N <= kMaxCapacity, "size must be less than size_type::max");
 
-    small_vector() : size_(0), capacity_(N) {}
+    small_vector() : size_(0), capacity_(N) { DebugUpdateWorkingStore(); }
 
     small_vector(const small_vector &other) : size_(0), capacity_(N) {
```
Comment (ncesario-lunarg) [inline layers/vk_layer_data.h:141]: We don't want `DebugUpdateWorkingStore()` for this constructor?

**Review by ncesario-lunarg — APPROVED** [review]
One question, but LGTM.

File: layers/vk_layer_data.h (hunk 1 above)
Comment (jzulauf-lunarg) [reply layers/vk_layer_data.h:141]: reserve calls it

Comment (ncesario-lunarg) [reply layers/vk_layer_data.h:141]: Ah, missed that. Thanks.

**Pull request merged**